.github/workflows/dockerimages.yml

```diff
@@ -6,7 +6,7 @@ on:
     - master
     paths:
     - scripts/docker/android-build-container/Dockerfile
-    - scripts/docker/mxe-build-container/Dockerfile
+#    - scripts/docker/mxe-build-container/Dockerfile
```

Comment:
can this be done in stages? I.e. create a first partial build and push it and use that as the base for a second part?
Awkward, but hey, you know - I like the underlying idea of this not just being something only I can do **quite a bit**

Reply:
I've thought about it, but I haven't gotten around to testing it, and after not getting around to it for long enough I'd just rather go with the bits that works, and leave the mxe-bits "as-is" to be continued on at some other time (or by someone else).

Reply:
That makes sense.
I'm curious, do you have a partial run of this (i.e., where was the process at the 6h mark)?

Reply:
(sorry, logged into the wrong account - I need to be Subsurface-divelog to play with the webhook, but I usually try to comment as "me")

Reply:
https://github.com/glance-/subsurface/runs/281752387

One if the big thingies is spending 3h building qtwebkit.

Reply:
> https://github.com/glance-/subsurface/runs/281752387
> 
> One if the big thingies is spending 3h building qtwebkit.

Thanks for sharing that. So to me this looks like there are three easy to identify logical blocks:

- create stage 1 container with qt
- create stage 2 container with qtwebkit
- create final container with everything ready

The mechanics to do that are very simple, I want to try that this weekend. The biggest question really is how to trigger this as actions so that we don't get bitten by the 6h limit without having to do very unnatural things to our repo. I'm reading up on the rules for GitHub actions to understand that better...

Reply:
So according to https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions the rule is "Each job in a workflow can run for up to 6 hours of execution time."
That should allow a *really* simple setup for this.
I'm not sure I'll have the time to try this out today, but I'll definitely find time over the weekend - this may be quite simple

Reply:
> So according to https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions the rule is "Each job in a workflow can run for up to 6 hours of execution time."
> That should allow a _really_ simple setup for this.
> I'm not sure I'll have the time to try this out today, but I'll definitely find time over the weekend - this may be quite simple

I did kick the test off last night and no, this is not simply a "per step" maximum - splitting things up is insufficient to do this. On the flip side I was able to have it push to Docker Hub, so at least one of the issues is solvable. The issue will be one of documentation as this requires the person playing with this to create their own docker hub account (and no, that does NOT have to have the same user name as your GitHub account) and put the credentials into secrets on GitHub.

Now I need to come up with a creative way around the time limit problem. 

I do wonder, thought, how much help this will bring towards updating these images as that is still a remarkably painful experience. And in many ways it is less painful doing this on your own machine as that is likely more powerful and therefore quicker than that GitHub Action VMs. So why would someone interested in helping do that as an Action in the first place instead of just doing that on their own laptop? But I guess the principle of having this happen "out in the open" instead of "behind closed doors on Dirk's machine" by itself is valuable.